lightning/src/ln/functional_test_utils.rs

```diff
@@ -1254,24 +1254,101 @@ macro_rules! get_route_and_payment_hash {
 	}}
 }
 
+pub struct HTLCHandlingFailedConditions {
+	pub expected_pending_htlcs_forwardable: bool,
+	pub expected_htlc_processing_failed: Option<usize>,
```

Comment:
FWIW, this field could be removed and instead use `expected_destinations.len()`. Maybe rename `expected_destinations` to `expected_htlc_processing_failures` or simply `expected_failures` if it is the only field. But if that's the case, could we just use a `Vec` instead of a struct?

Reply:
👍 done!